src/ast/query.rs

```diff
@@ -835,6 +843,24 @@ impl fmt::Display for TableAlias {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum TableVersion {
+    Timestamp(String),
+}
+
+impl Display for TableVersion {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            TableVersion::Timestamp(timestamp) => {
+                write!(f, " FOR SYSTEM_TIME AS OF '{timestamp}'")?
+            }
+        }
+        Ok(())
+    }
+}
```

Comment:
This may be a slight problem—some other systems have completely different syntaxes here but we're missing the info on which particular dialect the table version was parsed for.

For instance, Snowflake uses [`AT`](https://docs.snowflake.com/en/sql-reference/constructs/at-before), so if we decided to extend the support to it we wouldn't know how to re-construct the SQL here.

Reply:
One thing I wonder is whether the `AsOf` variant should be named explicitly as `ForSystemAsOf`, given that e.g. CockroachDB also has an [`AS OF` clause](https://www.cockroachlabs.com/docs/stable/as-of-system-time) with a somewhat different syntax. Arguably that dialect is not supported now but if someone decides to implement it it would be a misfortunate conflict.